### PR TITLE
HDDS-9196. Fix ECKeyOutputStream for secure S3

### PR DESCRIPTION
**Pull request by vtutrinov**
## What changes were proposed in this pull request?

Set S3 credentials for the thread to flush stripes (EC output stream)

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9196

## How was this patch tested?

manually, the test case is described in the Jira ticket

### CODE REVIEW

**Comment by kerneltime** [comment]
Thanks for working on this bug. OM should raise an exception when creating UGI. 
```
  public UserGroupInformation createUGI() {
```
A bad client should not lead to a NPE. Can you please add this to your PR?

**Comment by vtutrinov** [comment]
> Thanks for working on this bug. OM should raise an exception when creating UGI.
> 
> ```
>   public UserGroupInformation createUGI() {
> ```
> 
> A bad client should not lead to a NPE. Can you please add this to your PR?

Done

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMFinalizeUpgradeRequest.java [hunk 1]
```diff
@@ -67,7 +68,12 @@ public OMClientResponse validateAndUpdateCache(
 
     try {
       if (ozoneManager.getAclsEnabled()) {
-        final UserGroupInformation ugi = createUGI();
+        UserGroupInformation ugi;
+        try {
```
[the remark this answers is not in the document]
Comment (vtutrinov) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMFinalizeUpgradeRequest.java]: Done

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMFinalizeUpgradeRequest.java [hunk 2]
```diff
@@ -67,7 +68,12 @@ public OMClientResponse validateAndUpdateCache(
...
+          ugi = createUGI();
+        } catch (AuthenticationException e) {
+          throw new RuntimeException(e);
```
[the remark this answers is not in the document]
Comment (vtutrinov) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/upgrade/OMFinalizeUpgradeRequest.java]: Done

**Comment by vtutrinov** [comment]
@kerneltime could you review the PR again?

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @vtutrinov for the patch, @ivanzlenko, @kerneltime for the review.

**Pull request merged**